cypress/e2e/po/components/ember/ember-node-affinity.po.ts

```diff
@@ -0,0 +1,142 @@
+import EmberInputPo from '@/cypress/e2e/po/components/ember/ember-input.po';
+import EmberComponentPo from '@/cypress/e2e/po/components/ember/ember-component.po';
+import EmberNewSelectPo from '@/cypress/e2e/po/components/ember/ember-new-select.po';
+
+export default class EmberNodeAffinityPo extends EmberComponentPo {
+  findTerm(idx: number): NodeSelectorTermPo {
+    return new NodeSelectorTermPo(`${ this.selector } [data-testid="node-selector-term"]div:nth-child(${ idx + 1 })`);
+  }
+
+  /**
+   *
+   * @returns array of node affinity term elements, or null if none are defined
+   */
+  allTerms() {
+    return cy.iFrame().then((iframe) => {
+      const queryResult = iframe.find(`${ this.selector } [data-testid="node-selector-term"]`);
+
+      if (queryResult?.length > 0) {
+        return cy.iFrame().find(`${ this.selector } [data-testid="node-selector-term"]`);
+      }
+
+      return null;
+    });
+  }
+
+  /**
+   * Remove existing node affinity terms - does not assume any are defined
+   * @returns
+   */
+  removeAllTerms() {
+    return this.allTerms().then((terms) => {
+      if (terms) {
+        let idx = terms.length - 1;
+
+        while (idx >= 0) {
+          this.removeTerm(idx);
+          idx--;
+        }
+      }
+    });
+  }
+
+  removeTerm(idx: number) {
+    this.findTerm(idx).remove();
+  }
+
+  addTerm() {
+    return cy.iFrame().find(`${ this.selector } [data-testid="button-add-node-selector"]`).click();
+  }
+
+  // pod affinity po extends node affinity and adds a few fields in addition to _editTerm
+  editTerm(termData: any, idx: number) {
+    return this._editTerm(termData, idx);
+  }
+
+  /**
+   * Edit a node affinity term - assumes term at given index exists
+   * Priority and weight are overwritted if provided; match expressions/fields are appended
+   * @param termData
+   * @param idx
+   */
+  _editTerm(termData: any, idx: number) {
```

Comment:
Small tidyup. This can just be `editTerm`, editTerm from above can be removed, and editTerm in `ember-pod-affinity.po.ts` can call it via `super.editTerm`. As long as the `editTerm` functions are the same this should work.

If they differ at some point, `_editTerm` can become `private _editTerm` and be called in pod-affinity.po by `super._editTerm`.

Reply:
ohhh much tidier, thank you.  I've updated the pod/node affinity components accordingly.